googlemock/CMakeLists.txt

```diff
@@ -103,20 +103,28 @@ cxx_library(gmock_main
 # to the targets for when we are part of a parent build (ie being pulled
 # in via add_subdirectory() rather than being a standalone build).
 if (DEFINED CMAKE_VERSION AND NOT "${CMAKE_VERSION}" VERSION_LESS "2.8.11")
-  target_include_directories(gmock      SYSTEM INTERFACE "${gmock_SOURCE_DIR}/include")
-  target_include_directories(gmock_main SYSTEM INTERFACE "${gmock_SOURCE_DIR}/include")
+  set(gmock_include_dirs "${gmock_SOURCE_DIR}/include")
+  if (NOT "${CMAKE_VERSION}" VERSION_LESS "3.0.0")
+    set(gmock_include_dirs
+      $<BUILD_INTERFACE:${gmock_include_dirs}>
+      $<INSTALL_INTERFACE:include>)
+  endif()
+  target_include_directories(gmock      SYSTEM INTERFACE ${gmock_include_dirs})
```

Comment:
`SYSTEM` should be removed because `$<BUILD_INTERFACE:${gmock_include_dirs}>` will be used while building

Reply:
I think `PUBLIC` is better suitable here, instead of `INTERFACE`.

Reply:
@ruslo, I think you are right here. However, both the occurrence of `SYSTEM` and the use of `INTERFACE` predates this PR. I agree that they should be fixed, but:
* Not sure why `SYSTEM` is there. On my setups it works also without it but removing it might introduce issues with other compilers, i.e. it shall be discussed separately.
* AFAIK, to properly use $<BUILD_INTERFACE> the `include_directories()` occurrences should be replaced with `target_include_directories()`. But this task is not trivial because it has side effects on the tests. Also, it would require either additional CMake version switches or to update the minimum required version to something more recent than `2.6.4`. Personally, I would be in favor of the second, though I also understand it might have undesired side-effects I might not be aware of. AFAIK, also this shall be discussed in a separate PR.
* They are both not essential to the goal of this PR. Therefore, in the current PR proposal I dropped the `$<BUILD_INTERFACE>` occurrences because they were redundant with the existing `include_directories()` and only left the `$<INSTALL_INTERFACE>` which is required to complete this PR.

If you agree I would like to defer further changes to subsequent PRs.